File: AddressBook/react-client/src/dtos/ContactAddress.ts
import Common from "../utility/Common";
import { AddressLabelRes } from "./AddressLabel";
import { CityRes } from "./City";
import { PagedReq } from "./PagedReq";
import { ContactRes } from "./Contact";

export interface ContactAddressRes {
  contactAddressId?: string;
  line1?: string;
  line2?: string;
  postCode?: string;
  cityId?: string;
  city?: CityRes;
  addressLabelId?: string;
  addressLabel?: AddressLabelRes;
  contactId?: string;
  contact?: ContactRes;
}

export class ContactAddressReqEdit {
  line1?: string = "";
  line2?: string = "";
  postCode?: string = "";
  cityId?: string = "";
  addressLabelId?: string = "";
  contactId?: string = "";
  constructor(contactId?: string) {
    this.contactId = contactId;
  }
}

export class ContactAddressReqSearch extends PagedReq {
  contactId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy = "",
      searchText = "",
    }: PagedReq,
    {contactId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.contactId = contactId;
  }
}

File: AddressBook/Dtos/Dtos/ContactAddress.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class ContactAddressRes
    {
        public int ContactAddressId { get; set; }
        public string? Line1 { get; set; }
        public string? Line2 { get; set; }
        public string? PostCode { get; set; }

        // Foreign keys
        public int? CityId { get; set; }
        public CityRes? City { get; set; }

        public int? AddressLabelId { get; set; }
        public AddressLabelRes? AddressLabel { get; set; }

        public int? ContactId { get; set; }
        public ContactRes? Contact { get; set; }
    }

    public class ContactAddressReqEdit
    {
        [Required]
        public string? Line1 { get; set; }
        public string? Line2 { get; set; }
        public string? PostCode { get; set; }
        public int? CityId { get; set; }
        public int? AddressLabelId { get; set; }
        public int? ContactId { get; set; }
    }

    public class ContactAddressReqSearch : PagedReq
    {
        [Required]
        public int? ContactId { get; set; }
    }
}
